pil-analyzer/src/condenser.rs

```diff
@@ -1033,6 +1065,15 @@ fn try_value_to_expression<T: FieldElement>(value: &Value<'_, T>) -> Result<Expr
                     type_args: None,
                 }),
             ),
+
+            AlgebraicExpression::BinaryOperation(AlgebraicBinaryOperation { left, op, right }) => {
```

Comment:
I would suggest to put the full RHS of `Value::Expression(e) =>` into a function, because I could imagine the code then getting simpler and shorter.

And please return a `Result` instead of `unwrap()`ing it.